src/Symfony/Component/PropertyInfo/PropertyInfoExtractorCacheDecorator.php

```diff
@@ -0,0 +1,124 @@
+<?php
+
+/*
+ * This file is part of the Symfony package.
+ *
+ * (c) Fabien Potencier <fabien@symfony.com>
+ *
+ * For the full copyright and license information, please view the LICENSE
+ * file that was distributed with this source code.
+ */
+
+namespace Symfony\Component\PropertyInfo;
+
+use Doctrine\Common\Cache\Cache;
+
+/**
+ * Adds a cache layer on top of an extractor.
+ *
+ * @author Kévin Dunglas <dunglas@gmail.com>
+ */
+class PropertyInfoExtractorCacheDecorator implements PropertyInfoExtractorInterface
+{
+    /**
+     * @var PropertyInfoExtractorInterface
+     */
+    private $propertyInfoExtractor;
+
+    /**
+     * @var Cache
+     */
+    private $cache;
+
+    /**
+     * @var array
+     */
+    private $arrayCache = array();
```

Comment:
Yes the performance increase is signifiant: https://github.com/symfony/symfony/pull/16838#issuecomment-162169983

But :+1:  to add it in another decorator.